docs/manual/config/without-dm.rst

```diff
@@ -0,0 +1,64 @@
+====================
+Running from systemd
+====================
+
+This case will cover automatic login to Qtile after booting the system without
+using display manager directly from **systemd**. It logins in virtual console
+and init X by running through session.
+
+Automatic login to virtual console
+----------------------------------
+
+To get login into virtual console as an example edit `getty` service by running
+`systemctl edit getty@tty1` and add instructions:
+
+::
+    :caption: /etc/systemd/system/getty@tty1.service.d/override.conf
+
+    [Service]
+    ExecStart=
+    ExecStart=-/usr/bin/agetty --autologin username --noclear %I $TERM
+
+`username` should be changed to current user name.
+
+Check more for other `examples <https://wiki.archlinux.org/index.php/Getty#Automatic_login_to_virtual_console>`_.
+
+Qtile session
+-------------
+
+Starting Qtile directly will not preserve running applications on Qtile
+restarts by `qtile-cmd -o cmd -f restart` or other ways in Qtile itself. This
+can be solved by starting simple script loop.
+
+Autostart X session
+-------------------
+
+After login X session should be started. That can be done by `.bash_profile` if
+bash is used or `.zprofile` in case of zsh. Other shells can be adjusted by
+given examples.
+
+.. code-block:: bash
+
+    if systemctl -q is-active graphical.target && [[ ! $DISPLAY && $XDG_VTNR -eq 1 ]]; then
+      exec startx
+    fi
+
+And to start Qtile itself `.xinitrc` should be fixed:
+
+::
+
+    # some apps that should be started before Qtile, ex.
+    #
+    #   [[ -f ~/.Xresources ]] && xrdb -merge ~/.Xresources
+    #   ~/.fehbg &
+    #   nm-applet &
+    #   blueman-applet &
+    #   dunst &
+    #
+    # or
+    #
+    #   source ~/.xsession
+
+    while true; do
```

Comment:
I'm worried this could have nasty side effects, and I personally did not need this to run Qtile without DM and still have restarts working

Reply:
side effect could be that your qtile will broke but system will try to restart and will break again and so on

Reply:
Doesn't systemd have a respawn feature that could handle this more properly?

I personally do not use systemd, so I put this into `/etc/inittab` to accomplish the same:

```
c1:12345:respawn:/sbin/agetty --autologin <username> --noclear 38400 tty1 linux
```

(Note the `respawn` part, which will fallback to a simple shell if agetty crash-loops)

Reply:
Is it agetty or qtile that it will respawn on? The problem here is qtile dying and dropping you back to a shell, killing all your windows.

Reply:
agetty. But I'm not sure to understand: if qtile crashes then it makes sense to me that all applications get killed, and I don't see how this `while true` would change anything to that. Can you explain further?

Reply:
Oh sorry, I think I misread. Well, it will just respawn the `agetty --autologin`, so you'll end up in qtile again.

Reply:
Ugh. github destroyed my formatting, but you get the idea hopefully :)

Reply:
Which answers my latest comment in the other thread. Interesting!

Reply:
Your point stands, though, that the explanation here is wrong: qtile doing this on lazy.restart() is a bug, because we re-exec and don't actually exit. So the docs here need fixed.

Reply:
Hm, I'll recheck is Qtile restarting correctly and maybe it is enough for `exec qtile`. I had it before but I removed dm and got failed restarts. Thats why I created loop

Reply:
The failed restarts were a bug, but Qtile can have many bugs like this, so it's a worthwhile addition to the docs.

Reply:
But maybe better to add such story in separate block?

Reply:
I've have tested now without loop - all works. I'll update documentation to use `exec qtile`. And about loop, maybe I'll just write some sentence what you can do like that, but if we have config crash then we will loop in error